docs/sequence/PISPLinkingApi.puml

```diff
@@ -0,0 +1,654 @@
+deactivate PISP_TP_IN
+activate PISP_LM
+PISP_LM -> PISP_TP_OUT: return Accounts
+deactivate PISP_LM
+PISP_TP_OUT --> PISP: 200 OK Accounts
+
+
+note over PISP, DFSP_LM
+  ERROR_SCENARIO: Switch receives 7202 No accounts found for generic ID error.
+end note
+
+activate DFSP_LM
+DFSP_LM -> DFSP_LM: DFSPBackendRequests.getUserAccounts('username1234')
+DFSP_LM -> DFSP_LM: No accounts returned.
+DFSP_LM -> DFSP_LM: throw MojaloopFSPIOPError
+
+rnote right of DFSP_LM: state: errored
+rnote left of DFSP_LM #LightCoral
+PUT /accounts/username1234/error
+FSIOP-Source: dfspa
+FSIOP-Destination: pispa
+{
+  errorInformation: {
+    errorCode: '7202',
+    errorDescription: 'No accounts found for generic ID'
+  }
+}
+end note
+
+DFSP_LM -> Switch: PUT /accounts/username1234/error
+activate Switch
+Switch --> DFSP_LM: 200 OK
+deactivate DFSP_LM
+
+Switch --> PISP_TP_IN: PUT /accounts/username1234/error
+activate PISP_TP_IN
+PISP_TP_IN --> Switch: 200 OK
+deactivate Switch
+
+PISP_TP_IN -> PISP_LM: MojaloopFSPIOPError response recieved
+deactivate PISP_TP_IN
+activate PISP_LM
+PISP_LM -> PISP_TP_OUT: return MojaloopFSPIOPError
+rnote right of PISP_LM: state: errored
+deactivate PISP_LM
+PISP_TP_OUT --> PISP: 500 Internal Server Error ErrorInformationObject
+deactivate PISP_TP_OUT
+deactivate PISP
+
+== Request Consent - Web ==
+autonumber 1 "<b>REQUEST-CONSENT-#</b>"
+rnote right of PISP
+PISP presents accounts to user. User selects one or more accounts to link.
+end note
+PISP -> PISP_TP_OUT: POST /linking/request-consent
+rnote right of PISP #LightBlue
+POST /linking/request-consent
+{
+  "accounts": [
+    { accountNickname: "XXXXXXnt", id: "dfspa.username.1234", currency: "ZAR" },
+    { accountNickname: "SpeXXXXXXXXnt", id: "dfspa.username.5678", currency: "USD" }
+  ],
+  callbackURI: 'pisp-app://callback'
+}
+end note
+
+
+activate PISP
+activate PISP_TP_OUT
+
+PISP_TP_OUT -> PISP_TP_OUT: const model = await loadFromKVS()
+PISP_TP_OUT -> PISP_LM: model.requestConsent()
+
+activate PISP_LM
+
+rnote right of PISP_LM: state: requestConsent
+PISP_LM -> PISP_LM: ThirdpartyRequests.postConsentRequests()
+PISP_LM -> Switch: POST /consentRequests
+rnote right of PISP_LM #LightBlue
+POST /consentRequests
+FSIOP-Source: pispa
+FSIOP-Destination: dfspa
+{
+  // consentRequestId will be the uuid created
+  // at the PRELINKING-2
+  consentRequestId: 6789
+  scopes: [{
+    accountId: 'dfspa.username.1234',
+    actions: ['accounts.getBalance', 'accounts.transfer'],
+    accountId: 'dfspa.username.5678',
+    actions: ['accounts.getBalance', 'accounts.transfer'],
+  }],
+  // model will add `authChannels`
+  authChannels: ["WEB", "OTP"],
+  callbackURI: 'pisp-app://callback...'
+}
+end note
+
+activate Switch
+Switch --> PISP_LM: 202 Accepted
+deactivate PISP_LM
+
+Switch -> DFSP_TP_IN: POST /consentRequests
+activate DFSP_TP_IN
+DFSP_TP_IN --> Switch: 202 Accepted
+deactivate Switch
+
+DFSP_TP_IN -> DFSP_TP_IN: const model = await loadFromKVS()
```

Comment:
we are creating DFSPLinkinigModel here, we will call `saveToKVS` after receiving response form `fsm.requestConsent` transition (in `run` implementation)